### ISSUE TITLE
Saving and loading of the project

### ISSUE THREAD

**Issue by BeefEX** [bengsfort/tile-pusher#15]
I was thinking about how project saving should work. My idea is that save file could be some kind of archive, ZIP or any other, which would contain files with tilemap information as well as copies of all SpriteAtlas files that are needed. Another option would be saving relative path to the atlas files and not including them in the project. That would require the user to keep the same structure.

**Comment by bengsfort** [comment]
Definitely, I've been thinking about this as well. The reason for the classes such as `SpriteAtlasState` is so that we can have a serialisable state so we can re-create a session at load; but that still proves to be an issue when it comes to the 2D Textures (I admit I am not sure if they are serialisable; in which case we actually wouldn't need to do anything extra).

I know the way Tiled solves this is using relative paths and an xml-style file to store the overall app/map state. Then if a relative path cannot be found it will prompt you to find it to replace that reference; but here's an excerpt from one of those files:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<map . . . >
  <tileset firstgid="1" name="Dev-Environments" tilewidth="16" tileheight="16" tilecount="1024" columns="32">
    <image source="../../../Sprites/Environments_1.png" width="512" height="512" />
    . . .
  </tileset>
</map>
```

This is cool because it's an easily readable/parsable format, but it definitely has the downside that the user can't really share the map around without also sharing around the files (then again, you'd kind of expect that so you can make edits to your texture and have them auto-update....)

I'm unsure which is the _best_ way to go; but I do think that there is value in being able to send a file to your friend that allows them to open it up and get working immediately without any extra steps (like here is the texture, put it in ______ folder). But at the same time; it makes it so that the textures are not as easily updatable as we'll then have to have some sort of option in the UX to allow for updating a specific sprite sheet, whereas if we were to just store a relative path it would load up the latest version every time.

Thoughts? I probably just reiterated on everything you said and started rambling, in which case sorry :D

**Comment by BeefEX** [comment]
I think relative path are the way to go, it is much more flexible solution.

**Comment by BeefEX** [comment]
What about using relative paths and having `Export project` function that would pack the project and all required files into one project file for easy sharing?

**Comment by bengsfort** [comment]
Yeah, I agree.

And the `Export project` function would be a great addition! That would tick both of the boxes, maybe if you were to `Import project` it would then rebuild the project (and the assets) in the current directory. That would be really cool as it would enable people to have maximum flexibility in terms of updating assets but then they would still have the capability to bundle everything to send it to a collaborator.
